aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,729 @@
-.. include:: /../common/stub-topic.txt
+Razor Syntax Reference
+===========================
+`Taylor Mullen <https://twitter.com/ntaylormullen>`__, and `Rick Anderson`_
 
-|stub-icon| Razor Syntax
-========================
+.. contents:: Sections
+  :local:
+  :depth: 1
 
-.. include:: /../common/stub-notice.txt
+What is Razor?
+--------------
+Razor is a markup syntax for embedding server based code into web pages. The Razor syntax consists of Razor markup, C# and HTML. Files containing Razor generally have a *.cshtml* file extension.
 
-.. _issue: https://github.com/aspnet/Docs/issues/130
+.. contents:: Sections:
+  :local:
+  :depth: 1
+
+Rendering HTML
+--------------
+
+The default Razor language is HTML. Rendering HTML from Razor is no different than in an HTML file. A Razor file with the following markup:
+
+.. code-block:: none
+
+  <p>Hello World</p>
+
+Is rendered unchanged as ``<p>Hello World</p>`` by the server.
+
+Rendering server code
+----------------------
+
+Razor supports C# and uses the ``@`` symbol to transition from HTML to C#.
+Razor can transition from HTML into C# or into Razor specific markup. When an ``@`` symbol is followed by a :ref:`Razor reserved keyword <Razor-reserved-keywords-label>` it transitions into Razor specific markup, otherwise it transitions into plain C# .
+
+HTML containing ``@`` symbols may need to be escaped with a second ``@`` symbol. For example:
+
+.. code-block:: none
+
+ <p>@@Username</p>
+
+would render the following HTML:
+
+.. code-block:: none
+
+ <p>@Username</p>
+
+.. _razor-email-ref:
+
+``@`` in an email alias
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+HTML attributes containing email addresses don’t treat the ``@`` symbol as a transition character.
+
+ ``<a href="mailto:Support@contoso.com">Support@contoso.com</a>``
+
+Implicit Razor expressions
+---------------------------
+
+Implicit Razor expressions start with ``@`` followed by C# code. For example:
+
+.. code-block:: none
+
+  <p>@DateTime.Now</p>
+  <p>@DateTime.IsLeapYear(2016)</p>
+
+Implicit expressions generally cannot contain spaces. For example, in the code below, one week is not subtracted from the current time:
+
+.. literalinclude:: razor/sample/Views/Home/Contact.cshtml
+  :language: html
+  :start-after: @* End of greeting *@
+  :end-before: @*Add () to get correct time.*@
+
+Which renders the following HTML:
+
+.. code-block:: none
+
+  <p>
+    Last week: 7/7/2016 4:39:52 PM - TimeSpan.FromDays(7)
+  </p>
+
+Adding parenthesis fixes the problem:
```

Comment:
This seems repetitive with the following section on explicit expressions. How about just adding forward link in a note/tip? Like this:

``` rst
.. tip:: You can use C# expressions in Razor that contain spaces using an :ref:`explicit expression <explicit-razor-expressions>`
```